04/app.js

```diff
@@ -1,89 +1,105 @@
-import React from 'react';
-import { createRoot } from 'react-dom/client';
+import React from "react";
+import { createRoot } from "react-dom/client";
 
-const root = createRoot(document.querySelector('#root'));
+const root = createRoot(document.querySelector("#root"));
 
 class App extends React.Component {
-    state = { 
-        firstName: '',
-        lastName: '',
-        searchQuery: '',
-        users: ['Jan Kowalski', 'Michał Nowak'],
-    }
+  state = {
+    firstName: "",
+    lastName: "",
+    searchQuery: "",
+    users: ["Jan Kowalski", "Michał Nowak"],
+  };
 
-    renderUsersList() {
-        const {users} = this.state;
-        return users.map(name => {
-            return (
-                <li onClick={ this.clickHandler }>
-                    { name }
-                </li>
-            );
-        });
-    }
+  generateKey = (pre) => {
+    return `${pre}_${new Date().getTime()}`;
+  };
```

Comment:
Czy faktycznie generowanie `key` w ten sposób nam coś daje?
`key` jest wykorzystywane do tego, aby React wiedział czy ten element/komponent ma być ponownie przebudowany (czy ma być wykorzystany cache). Jeśli za każdym renderem generujemy "losowy" `key` to mija się to z celem :) Takie rozwiązanie byłoby ok, gdybyś do danych obieranych z inputa dodawał tą informację.